docs/design-proposals/2025-11 Gogo Protobuf.md

```diff
@@ -0,0 +1,53 @@
+---
+Authors: Mariell Hoversholm (@Proximyst)
+Created: 2025 November
+Last updated: 2025-11-25
+---
+
+# Usage of `gogo/protobuf`
+
+## Summary
+
+This is a retroactive document, intended to document a decision that has already
+been made.
+
+We use a fork of Protobuf, [`gogo/protobuf`], rather than the official
+`google.golang.org/protobuf` or `github.com/golang/protobuf` libraries.
+This is fuelled by one major reason: the official libraries don't behave well
+enough in terms of performance. This is a decision that we inherited from
+OpenTelemetry, see also [this issue](https://github.com/open-telemetry/opentelemetry-collector/issues/7095).
+
+The OpenTelemetry issue linked cites the large amount of memory allocations by
+the generated code from the official libraries as a particularly important
+reason for this decision. This is seen in generated code as use of `[]*Value`
+pointer-values instead of `[]Value` value-types.
+
+## Alternatives
+
+As of writing, there are many alternatives that exist, to varying degrees of
+sufficiency. When we look at migration, read up on the following projects, as
+well as anything that is relevant at that future time:
+
+- [vtprotobuf](https://github.com/planetscale/vtprotobuf)
+- [csproto](https://github.com/CrowdStrike/csproto)
+- [hyperpb](https://github.com/bufbuild/hyperpb-go)
+- [pdatagen](https://github.com/open-telemetry/opentelemetry-collector/tree/974da01f71487422c02fadadb8f66147162fcb14/internal/cmd/pdatagen)
```

Comment:
i don't think this should be included as it doesn't actually handle proto encoding/decoding. it is incapable of replacing gogoproto